backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -261,6 +272,25 @@ public RequestStatus updateProject(Project project, User user) throws SW360Excep
         }
     }
 
+    private boolean isDependenciesExists(Project project) {
+        boolean validReleaseIds = true, validLinkedProjectIds = true, validObligationId = true;
+        if (project.isSetReleaseIdToUsage()) {
+            Set<String> releaseIds = project.getReleaseIdToUsage().keySet();
+            validReleaseIds = DatabaseHandlerUtil.isAllIdInSetExists(releaseIds, releaseRepository);
+        }
+
+        if (project.isSetLinkedProjects()) {
+            Set<String> projectIds = project.getLinkedProjects().keySet();
+            validLinkedProjectIds =  DatabaseHandlerUtil.isAllIdInSetExists(projectIds, repository);
+        }
+
+        if (project.isSetLinkedObligationId()) {
```

Comment:
Maybe it would be good to stop validation once any doc id is not found and throw error message.
Same for validate Component and Release

If possible we can check is Linked Projects are Visible to user using `PermissionUtils.makePermission(project, user).isActionAllowed(RequestedAction.READ)`

Reply:
Done